internal/pkg/db/redis/data.go

```diff
@@ -243,7 +243,7 @@ func (c *Client) ReadingsByDeviceAndValueDescriptor(deviceId, valueDescriptor st
 	objects, err := getObjectsByRangeFilter(conn,
 		db.ReadingsCollection+":device:"+deviceId,
 		db.ReadingsCollection+":name:"+valueDescriptor,
-		0, limit-1)
+		0, int64(limit-1))
```

Comment:
Is this due to a concern that the `int` type may not supply a large enough value? There's no way we're going to send back 2.1 billion readings via a REST response, let alone more than that.

Reply:
I was reacting to changes that came out of the notification service support. I'll revisit and follow up.

Reply:
Looks like it was propagation from the notification interface. I've corrected to an `int` implementaiton.